src/libs/ajax.js

```diff
@@ -58,10 +74,15 @@ const query = (url, data, method = 'GET', options = {}) => {
     };
     if (method === 'GET') {
       xhr.open(method, `${url}?${dataToUrl(data)}`);
+      setHeaders(xhr, headers);
       xhr.send();
     } else {
       xhr.open(method, url);
-      xhr.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+      setHeaders(xhr, headers);
+      if (method === 'POST' && !headersContain(headers, 'Content-Type') &&
+          data && data.length > 0) {
```

Comment:
`data` is an object (that will be stringified below), so using `data.length` is incorrect here.